aviary/subsystems/propulsion/prop_performance.py

```diff
@@ -195,6 +178,59 @@ def setup(self):
                 promotes=['*'],
             )
 
+        self.add_subsystem(
+            'temperature_term',
+            om.ExecComp(
+                'theta_T = T0 * (1 + .2*mach**2)/T_amb',
+                theta_T={'units': "unitless", 'shape': nn},
+                T0={'units': 'degR', 'shape': nn},
+                mach={'units': 'unitless', 'shape': nn},
+                T_amb={'val': np.full(nn, 518.67), 'units': 'degR'},
+                has_diag_partials=True,
+            ),
+            promotes_inputs=[
+                ('T0', Dynamic.Mission.TEMPERATURE),
+                ('mach', Dynamic.Mission.MACH),
+            ],
+            promotes_outputs=['theta_T'],
+        )
+
+        self.add_subsystem(
+            'prop_tip_spd',
+            om.ExecComp(
+                'prop_tip_speed = minimum(pc_rotor_rpm_corrected * theta_T**.5 * max_prop_tip_spd, max_prop_tip_spd)',
```

Comment:
Do you expect the this minimum function to switch between the expression or the `max_prop_tip_spd` frequently during an optimization? If so, using a [smooth minimum](https://en.wikipedia.org/wiki/Smooth_maximum) function would improve convergence.

Based on the calculations here, I wouldn't expect using a regular minimum function to be a problem in this case. This is more of a general note and something to consider when using gradient-based optimization.

Reply:
For takeoff, I expect that it is always at maximum tip speed. For cruise, I expect that it is always less than maximum tip speed. I use `maximum()` because it is a built-in function for `ExecComp`. I am not sure if a smooth function can replace it.